PC/winreg.c

```diff
@@ -2081,35 +2119,21 @@ inskey(PyObject * d, char * name, HKEY key)
 
 #define ADD_KEY(val) inskey(d, #val, val)
 
-
-static struct PyModuleDef winregmodule = {
-    PyModuleDef_HEAD_INIT,
-    "winreg",
-    module_doc,
-    -1,
-    winreg_methods,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-PyMODINIT_FUNC PyInit_winreg(void)
+static int
+exec_module(PyObject *m)
 {
-    PyObject *m, *d;
-    m = PyModule_Create(&winregmodule);
-    if (m == NULL)
-        return NULL;
+    winreg_state *st = (winreg_state *)_PyModule_GetState(m);
+
+    PyObject *d;
     d = PyModule_GetDict(m);
-    PyHKEY_Type.tp_doc = PyHKEY_doc;
-    if (PyType_Ready(&PyHKEY_Type) < 0)
-        return NULL;
+    st->PyHKEY_Type = (PyTypeObject *)
+                       PyType_FromModuleAndSpec(m, &pyhkey_type_spec, NULL);
```

Comment:
```suggestion
    st->PyHKEY_Type = (PyTypeObject *)
                       PyType_FromModuleAndSpec(m, &pyhkey_type_spec, NULL);
    if (st->PyHKEY_Type == NULL)
        return -1;
```

Should we prevent this situation?

Reply:
Yes, of course. Good catch!